aci/resource_aci_rtctrlattrp.go

```diff
@@ -128,6 +178,35 @@ func resourceAciActionRuleProfileCreate(ctx context.Context, d *schema.ResourceD
 		return diag.FromErr(err)
 	}
 
+	// setRouteTagDn - Operations
+	if setRouteTag, ok := d.GetOk("set_route_tag"); ok {
+
+		log.Printf("[DEBUG] RtctrlSetTag: Beginning Creation")
+
+		rtctrlSetTagAttr := models.RtctrlSetTagAttributes{}
+		rtctrlSetTagAttr.Tag = setRouteTag.(string)
+		rtctrlSetTag := models.NewRtctrlSetTag(fmt.Sprintf(models.RnrtctrlSetTag), rtctrlAttrP.DistinguishedName, "", "", rtctrlSetTagAttr)
+
+		creation_err := aciClient.Save(rtctrlSetTag)
+		if creation_err != nil {
+			return diag.FromErr(creation_err)
+		}
+
+		log.Printf("[DEBUG] %s: Creation finished successfully", rtctrlSetTag.DistinguishedName)
+		resourceAciRtctrlSetTagRead(ctx, rtctrlSetTag.DistinguishedName, d, m)
+
+	} else {
+		setRouteTagDn := rtctrlAttrP.DistinguishedName + fmt.Sprintf("/"+models.RnrtctrlSetTag)
+		log.Printf("[DEBUG] %s: rtctrlSetTag - Beginning Destroy", setRouteTagDn)
+
+		err := aciClient.DeleteByDn(setRouteTagDn, "rtctrlSetTag")
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		log.Printf("[DEBUG] %s: rtctrlSetTag - Destroy finished successfully", setRouteTagDn)
+	}
```

Comment:
I don't think you need to take care of this in the create function. This is more for an update case as if it is a Create it should be a brand new parent without any children.